server/vertx/src/main/scala/sttp/tapir/server/vertx/routing/PathMapping.scala

```diff
@@ -0,0 +1,50 @@
+package sttp.tapir.server.vertx.routing
+
+import io.vertx.core.http.HttpMethod
+import io.vertx.scala.ext.web.{Route, Router}
+import sttp.tapir.{Endpoint, EndpointInput}
+import sttp.tapir.EndpointInput.PathCapture
+import sttp.tapir.internal._
+
+object PathMapping {
+
+  // The necessary stuff to create a Route from a Router
+  type RouteDefinition = (Option[HttpMethod], String)
+
+  /**
+   * Given a Router, creates a Vert.x Route matching the route definition
+   * @param router a Vert.x Router
+   * @param route the definition of the route (method, and path)
+   * @return a route, attached to the router
+   */
+  private [vertx] def createRoute(router: Router, route: RouteDefinition): Route =
+    route match {
+      case (Some(method), path) => router.route(method, path)
+      case (None, path)         => router.route(path)
+    }
+
+  /**
+   * Extracts the route definition from the endpoint inputs
+   * @param endpoint a Tapir endpoint
+   * @return the route definition matching the endpoint input definition
+   */
+  private [vertx] def extractRouteDefinition(endpoint: Endpoint[_, _, _, _]): (Option[HttpMethod], String) =
+    (MethodMapping.sttpToVertx(endpoint.httpMethod), extractVertxPath(endpoint))
+
+  private def extractVertxPath(endpoint: Endpoint[_, _, _, _]): String = {
```

Comment:
maybe `endpoint.renderPathTemplate` would work here instead?

Reply:
I tried to at some point, and it failed.
I'll give it another try now that every test is working, to understand a bit more clearly why it fails.

Reply:
`renderPathTemplate`  doesn't work, because it's missing the `/*` and thus does not "catches" some endpoints.
Example of a failing test: "no path should match anything".

I'll stick to the old implementation (taken from Finatra I guess).